src/overlays/actors/ovl_En_Kakasi3/z_en_kakasi3.c

```diff
@@ -270,29 +271,32 @@ void func_80A915B8(EnKakasi3* this, GlobalContext* globalCtx) {
 void func_80A91620(EnKakasi3* this, GlobalContext* globalCtx) {
     Player* player = PLAYER;
 
-    if (globalCtx->msgCtx.unk_E3EE == 4 || (globalCtx->msgCtx.unk_E3EE >= 5 && globalCtx->msgCtx.unk_E3EE < 0xB)) {
+    if ((globalCtx->msgCtx.unk_E3EE == 4 || (globalCtx->msgCtx.unk_E3EE >= 5 && globalCtx->msgCtx.unk_E3EE < 11)) &&
+        (globalCtx->msgCtx.msgMode == 0)) {
 
-        if (globalCtx->msgCtx.msgMode == 0) {
-            func_800803F0(globalCtx, this->camId);
-            if (globalCtx->cameraPtrs[this->camId] == NULL) {
-                this->camId = -1;
-            }
-            if (this->camId != -1) {
-                func_8005B1A4(globalCtx->cameraPtrs[this->camId]);
-            }
-            this->actionFunc = func_80A911F0;
-            return;
+        func_800803F0(globalCtx, this->camId);
+        if (globalCtx->cameraPtrs[this->camId] == NULL) {
+            this->camId = -1;
         }
+        if (this->camId != -1) {
+            func_8005B1A4(globalCtx->cameraPtrs[this->camId]);
+        }
+        this->actionFunc = func_80A911F0;
+        return;
     }
 
     if (globalCtx->msgCtx.unk_E3EE == 3 && globalCtx->msgCtx.msgMode == 0) {
         this->dialogState = 5;
         func_8010B680(globalCtx, 0x40A5, NULL);
         func_8002DF54(globalCtx, NULL, 8);
         this->actionFunc = func_80A91A90;
-    } else if (globalCtx->msgCtx.unk_E3EE == 1) {
+        return;
+    }
+
+    if (globalCtx->msgCtx.unk_E3EE == 1) {
         func_80A90EBC(this, globalCtx, 0);
         player->stateFlags2 |= 0x800000;
+        return;
```

Comment:
no need to return; at the end of the function, and also `func_80A90EBC` isn't an action setup

sorry if I'm being annoying with this 😅

Reply:
cleaning up stuff like this at this stage is good, needs to be done :D
after this change this is good to for me

Reply:
don't worry it's fine, it's hard to keep apply the suggestions across all 3 PR's 😅 

What does imply that the function isn't a action setup?

Reply:
What I call "action setup" is anything that does `this->actionFunc = ...;`, often along with setting a bunch of struct members
`func_80A90EBC` doesn't do that anywhere, its purpose seems to be reusing common logic from different actions but that's up to further documentation

Unrelated but avoid force-pushing please, it ruins tracking of comments and commits by github

Reply:
Ah, ok thanks for the clarification.

Yeah I've just noticed about the force push